crates/forge-runner/src/running.rs

```diff
@@ -304,12 +295,15 @@ fn extract_test_case_summary(
     match run_result {
         Ok(result_with_info) => {
             match result_with_info.run_result {
-                Ok(run_result) => Ok(TestCaseSummary::from_run_result_and_info(
-                    run_result,
-                    case,
-                    args,
-                    result_with_info.gas_used,
-                )),
+                Ok(run_result) => {
+                    let summary = TestCaseSummary::from_run_result_and_info(
+                        run_result,
+                        case,
+                        args,
+                        result_with_info.gas_used,
+                    );
+                    Ok(check_available_gas(&case.available_gas, summary))
```

Comment:
Imo `check_available_gas` this should be used inside `from_run_result_and_info`, this way we don't leak the logic

Reply:
Summoning @Arcticae

Reply:
Okay, I see @Arcticae's comment and I just disagree xD Even the name `from_run_result_and_info` suggests that it retrieves the result from run result and *additional info* like gas used. Otherwise it is just confusing - why would you construct the result and the change it randomly?

Reply:
I could go both ways but i always tend to leave business logic out of constructors (just a habit that i have, and it produces clean code most of the times). We could go with a builder as well. But i really do have vietnam flashbacks when it comes to builders.

Reply:
@war-in leaving the decision up to you.